### ISSUE TITLE
regularized_eigh backward: wrong cotangent convention for complex Hermitian input

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#753]
Found while fixing #750/#751. `regularized_eigh`'s backward shares the complex cotangent-convention defect described in #751b.

## Claim

`_regularized_eigh_bwd` in `src/tenax/algorithms/_ad_primitives.py` assembles the textbook eigh adjoint, whose cotangents are `dL/dX`. JAX's cotangents for a real-valued loss of a complex input are the conjugate of that. For **complex Hermitian** input with an eigen*vector*-dependent loss, the returned gradient is wrong in direction.

The F-matrix orientation here was already corrected in #316, so the **real** path is fine — this is complex-only.

## Evidence

AD vs finite differences, 6×6 Hermitian, loss `Re⟨W_h, V diag(w²) Vᴴ⟩` (gauge-invariant), with `jnp.linalg.eigh`'s VJP as the control:

| input | `regularized_eigh` | `jnp.linalg.eigh` (control) |
|---|---|---|
| real symmetric | rel 7.5e-09, cos 1.000000000 | rel 7.5e-09, cos 1.000000000 |
| **complex Hermitian** | rel **1.63e+00**, cos **0.122** | rel 1.06e-08, cos 1.000000000 |

## Why the existing tests miss it

An eigen*value*-only loss passes even with the bug:

| loss | complex Hermitian, buggy |
|---|---|
| `sum(w)` | rel 5.9e-09, cos 1.000000000 |
| `Re⟨W_h, V w² Vᴴ⟩` | rel 1.63e+00, cos 0.122 |

`d(Σw)/dM = I`, which is real, so a missing conjugation is invisible. Any gate built on eigenvalue sums is structurally blind to this.

## Fix

The same conj-in/conj-out bridge used for the SVD kernel:

```python
dw = dw.conj()
dv = dv.conj()
...
return (dM.conj(),)
```

Conjugation commutes with the Hermitian symmetrisation, so placement relative to `dM = 0.5*(dM + dM.conj().T)` is free. `conj` is the identity on real dtypes, so the real path is bit-identical.

After: complex `rel = 1.06e-08, cos = 1.000000000` — matching JAX's control exactly.

## Blast radius

`regularized_eigh` backs the eigh-based CTM projector path (`projector_method="eigh"`). Complex-valued states only; real-valued Hamiltonians are unaffected.

Fixed together with #750/#751. Gated by `test_eigh_eigenvector_gradient_matches_finite_differences` in `tests/test_svd_adjoint_fd_750.py`, with the eigenvalue-only companion kept alongside to document why it cannot stand alone.

Related: #316 (F-matrix orientation in the same function), #750, #751.

> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.
